cloud/terraform/machineactuator.go

```diff
@@ -351,10 +353,110 @@ func (tf *TerraformClient) PostDelete(cluster *clusterv1.Cluster, machines []*cl
 }
 
 func (tf *TerraformClient) Update(cluster *clusterv1.Cluster, goalMachine *clusterv1.Machine) error {
-	glog.Infof("TERRAFORM UPDATE.\n")
+	// Try to get the annotations for the versions. If they don't exist, update the annotation and return.
+	// This can only happen right after bootstrapping.
+	if goalMachine.ObjectMeta.Annotations == nil {
+		ip, _ := tf.GetIP(goalMachine)
+		glog.Info("Populating existing state for bootstrapped machine.")
+		return tf.updateAnnotations(goalMachine, ip)
```

Comment:
Why does the return here? And why is Annotations being nil a condition for this? A user can create a master with annotations already.

Reply:
> Why does the return here

After bootstrap and the pivot, the controller in the cluster will see this object as an update (so this condition only ever hits after bootstrap). In GCE controller, we handle that by adding GCP labels.

> And why is Annotations being nil a condition for this? A user can create a master with annotations already.

That's true. I'll check for the specific key I care about as a guard here.